### ISSUE TITLE
Expose an API to check if there are pending crash reports

### ISSUE THREAD

**Issue by monotkate** [getsentry/react-native-sentry#322]
If I'm missing somewhere it's already doing this, please let me know.  We would like to know when booting the app if there are any crashes that occurred on the previous run, and if there is a pending report to be sent.  This allows us to take action to wipe user data and prevent another crash.

- `setShouldSendCallback` runs before the crash, and there isn't enough time to record anything to storage
- `setEventSentSuccessfully` runs after the report has been sent, which is after our boot workflow has completed

We used to have manual checks in place using Raven, but I don't believe the endpoints are exposed anymore.

**Comment by HazAT** [comment]
Hey, so this is not easy to do in react-native since we currently only know on the native side if we crashed or not.
There is a native function called `crashedLastLaunch` which returns a boolean 
but
I suppose it will not work because in order to get the value from it you have to call `install()` in react-native and communicate over the javascript bridge. 
Meanwhile, any pending reports would have been already sent.

So I am not really sure if we can reliably solve this issue like this.

**Comment by monotkate** [comment]
When are the pending reports sent then, during install?

**Comment by HazAT** [comment]
Yes, basically.

**Comment by JAStanton** [comment]
@HazAT  Will `crashedLastLaunch` return `false` if all pending reports are sent?

**Comment by HazAT** [comment]
No, `crashedLastLaunch` will stay `true`.

**Comment by JAStanton** [comment]
@HazAT why do you not recommend using `crashedLastLaunch` then to determine if crashed on last boot? 

Our app workflow would be:
- Begin booting the app
- call `codepush#install`
- expose `crashesLastLaunch` to native bridge and call new `codepush#didCrashLastLaunch` to see if last launch did crash
- if true do A if false do B

**Comment by monotkate** [comment]
Is `crashedLastLaunch` ios only, or does it exist for android as well?

**Comment by HazAT** [comment]
So this flag is not (yet) available on Android :/
@JAStanton Keep in mind, `crashedLastLaunch` is not blocking nor does it have a callback, you would need to call `install` and wait until it's ready, not sure if that would be a good practice to "delay" the app start.
I guess what you are trying to achieve is currently not possible at least in a convenient way.
I will discuss this issue with my colleagues and post the outcome here.

**Comment by JAStanton** [comment]
@HazAT thanks I'd love to know the outcome of that discussion!

The use case / context is that we are trying to get our users to be upgraded to the latest version of CodePush at all costs. So say they boot the app, there is a pending update, it tries to apply the update but then update explodes and crashes, on next boot it will rollback to previous version. Ideally on that boot we detect there was a previous crash, wipe all users state and try to apply the update again. 

I get what you're saying about delaying the app boot while we wait for install to finish. If it's less than a second or maybe even a full second or two we may consider it worth waiting for. This workflow is pretty valuable for us so we would be willing to delay the user if it means higher success rate of updating to be on the latest app version.

We forked and added the necessary iOS piece, but we were hoping it wouldn't be that much more difficult to add the Android piece. We're waiting on you to let us know if we're walking into a big trap or not :)

**Comment by monotkate** [comment]
Got this working locally by waiting for `install` to complete and it doesn't seem to delay the app's boot significantly.  Will need to test further to see what the actual delay may be.  Is it something you would want us to PR back to sentry, or just fork and use?

**Comment by HazAT** [comment]
Hey,
Sorry for the late reply but we think for now your solution for iOS will work just fine 👍 
If you don't mind submitting a PR to our repo, that would really help.

In regards to Android, the SDK itself doesn't provide a way to find out if a crash happened on the last startup. 
I will find a way to add it there, e.g.: checking in `addShouldSendEventCallback` and set a var if it is a `JavascriptException`.

**Comment by monotkate** [comment]
Here's the PR: https://github.com/getsentry/react-native-sentry/pull/329

**Comment by JAStanton** [comment]
@HazAT should we file a separate issue to get this in Android?

**Comment by HazAT** [comment]
@JAStanton That would be awesome.
I think though, it might be trickier there since I believe we don't provide an API if a crash occurred last start.
